src/rest_vol.c

```diff
@@ -1539,7 +1539,8 @@ H5_rest_url_encode_path(const char *_path)
     if (!_path)
         FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, NULL, "path was NULL");
 
-    path = (char *)_path;
+    /* Silence compiler const warnings */
+    memcpy(&path, &_path, sizeof(char *));
```

Comment:
This works (if a bit awkward), although we could probably get away with just making `path` and `cur_pos` const while using a different var. than `cur_pos` when indexing into `tmp_buffer`.